File: app/src/main/java/com/websarva/wings/android/zuboradiary/ui/list/diarylist/DiaryDayListItem.java
package com.websarva.wings.android.zuboradiary.ui.list.diarylist;

import androidx.annotation.NonNull;

import com.websarva.wings.android.zuboradiary.data.database.DiaryListItem;
import com.websarva.wings.android.zuboradiary.ui.list.DiaryDayListBaseItem;

public class DiaryDayListItem extends DiaryDayListBaseItem {

    private final String title;
    private final String picturePath;

    DiaryDayListItem(DiaryListItem listItem) {
        super(listItem);

        this.title = listItem.getTitle();
        this.picturePath = listItem.getPicturePath();
    }

    @NonNull
    public String getTitle() {
        return title;
    }

    @NonNull
    public String getPicturePath() {
        return picturePath;
    }

}

File: app/src/main/java/com/websarva/wings/android/zuboradiary/ui/list/diarylist/DiaryDayListItem.kt
package com.websarva.wings.android.zuboradiary.ui.list.diarylist

import android.net.Uri
import com.websarva.wings.android.zuboradiary.data.database.DiaryListItem
import com.websarva.wings.android.zuboradiary.ui.list.DiaryDayListBaseItem

internal class DiaryDayListItem(listItem: DiaryListItem) :
    DiaryDayListBaseItem(listItem) {

    val title: String = listItem.title
    val picturePath: Uri?

    init {
        val picturePath = listItem.picturePath
        if (picturePath.isEmpty()) {
            this.picturePath = null
        } else {
            this.picturePath = Uri.parse(picturePath)
        }
    }
}
